### ISSUE TITLE
Not really an issue, just wondering interest value

### ISSUE THREAD

**Issue by GregSilverman** [dpgaspar/Flask-AppBuilder#447]
For a project I've been using Flask Appbuilder with to create a patient registry one of the requirements was to extend the built-in search widget to allow for multiple searches on the same field (e.g., a list of Boolean ANDed diagnoses by patient). To do this, I had to extend the query method in the SQLAinterface to allow for arbitrary numbers of self-joins across a model instance. I will make this available as soon as I clean up the code.

Not sure if you would want to merge into your code base, or if I should just fork this. Thoughts?

**Comment by dpgaspar** [comment]
Not sure i'm following this, you can already search multiple times the same field, they will all be ANDed.

Can you be more detailed?

**Comment by GregSilverman** [comment]
Briefly, without going into too much detail, as an example, if I have the class:

`Diagnosis(Model):
    __tablename__ = 'vw_svc_diagnosis'
    diagnosis_id = Column(String(32), primary_key=True)
    dx_name = Column(String(255))
    mrn = Column(String(255))
    dx_code = Column(String(255))`

And I want to query for the set of patients (by MRN) who have 
`dx_name = 'Diagnosis 1' AND dx_name = 'Diagnosis 2' AND dx_name = 'Diagnosis 3'`, etc., out of the box, this will not work, since 

a) get_filter_args in urtools.py and all the functions in filters.py expect a dictionary and thus will not allow multiple filters of the same type on the same attribute

b) to get the above query, I need to do a JOIN across the Diagnosis class linked by the patient MRN.

In any case, I have circumvented this by appropriate overrides in the methods and modules noted above and also in the query method of the sqla/interface.py module.

I am working on trying to generalize it so that it is not so specific to the application for which I had to do this. I will gladly share the code, once it is ready.
